src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1211,16 +1223,55 @@ namespace xclcpuemhal2 {
       return;
     }
 
+    setDeviceProcessStarted(false);
     std::string socketName = sock->get_name();
-    if(socketName.empty() == false)// device is active if socketName is non-empty
+    // device is active if socketName is non-empty
+    if (!socketName.empty())
     {
 #ifndef _WINDOWS
       xclClose_RPC_CALL(xclClose,this);
 #endif
     }
+   closeMessengerThread();
    saveDeviceProcessOutput();
   }
 
+  void CpuemShim::closeMessengerThread()
+  {
+    if (mMessengerThread.joinable())
+      mMessengerThread.join();
+  }
+
+  void CpuemShim::messagesThread()
+  {
+    auto start_time = std::chrono::high_resolution_clock::now();
+    auto lpath = this->deviceDirectory + "/../../../device_process.log";
+    sParseLog deviceProcessLog(lpath);
+    int count = 0;
+    while (getDeviceProcessStarted())
+    {
+      // I may not get ParseLog() all the times, So Let's optimize myself.
+      // Let's sleep for 10,20,30 seconds....etc until it is < 300 seconds
+      // After that I may not get extensive parseLog() statements.
+      // So I want to call it for each interval of 300 seconds
+
+      auto end_time = std::chrono::high_resolution_clock::now();
+
+      if (std::chrono::duration_cast<std::chrono::seconds>(end_time - start_time).count() <= 300) {  
```

Comment:
Being that the magic number "300" is used multiple times, it might be best to have this be assigned to a constexpr variable and use it instead.